docs/modules/ROOT/pages/erc721.adoc

```diff
@@ -0,0 +1,86 @@
+= ERC-721
+
+We've discussed how you can make a _fungible_ token using xref:erc20.adoc[ERC-20], but what if not all tokens are alike?
+This comes up in situations like *real estate*, *voting rights*, or *collectibles*, where some items are valued more than others, due to their usefulness, rarity, etc.
+ERC-721 is a standard for representing ownership of xref:tokens.adoc#different-kinds-of-tokens[_non-fungible_ tokens], that is, where each token is unique.
+
+ERC-721 is a more complex standard than ERC-20, with multiple optional extensions, and is split across a number of contracts.
+The OpenZeppelin Contracts provide flexibility regarding how these are combined, along with custom useful extensions.
+Check out the https://docs.rs/openzeppelin_stylus/token/erc721/struct.Erc721.html[`API reference`] to learn more about these.
+
+== Constructing an ERC-721 Token Contract
+
+We'll use ERC-721 to track items in our game, which will each have their own unique attributes.
+Whenever one is to be awarded to a player, it will be minted and sent to them.
+Players are free to keep their token or trade it with other people as they see fit, as they would any other asset on the blockchain!
+Please note any account can call `awardItem` to mint items.
+To restrict what accounts can mint items we can add xref:access-control.adoc[Access Control].
+
+Here's what a contract for tokenized items might look like:
+
+[source,rust]
+----
+sol_storage! {
+    #[entrypoint]
+    struct GameItem {
+        #[borrow]
+        Erc721 erc721;
+        #[borrow]
+        Metadata metadata;
+        #[borrow]
+        UriStorage uri_storage;
+        uint256 _next_token_id;
+    }
+}
+
+#[external]
+#[inherit(Erc721, Metadata)]
+impl GameItem {
+    pub fn award_item(
+        &mut self,
+        player: Address,
+        token_uri: String,
+    ) -> Result<U256, Vec<u8>> {
+        let token_id = self._next_token_id.get() + uint!(1_U256);
+        self._next_token_id.set(token_id);
+
+        self.erc721._mint(player, token_id)?;
+        self.uri_storage._set_token_uri(token_id, token_uri);
+
+        Ok(token_id)
+    }
+
+    #[selector(name = "tokenURI")]
+    pub fn token_uri(&self, token_id: U256) -> Result<String, Vec<u8>> {
```

Comment:
I've noticed that the example similar to solidity one looks complex for documentation